resolwe/flow/models.py

```diff
@@ -372,6 +374,15 @@ def save(self, render_name=False, *args, **kwargs):
         if not self.descriptor:
             render_descriptor(self)
 
+        if create:
+            validate_schema(self.input, self.process.input_schema)
+        if self.descriptor_schema:
+            validate_schema(self.descriptor, self.descriptor_schema.schema)
+        elif self.descriptor and self.descriptor != {} and self.descriptor != "{}":
+            raise ValueError("`descriptor_schema` must be defined if `descriptor` is given")
+        if self.status == Data.STATUS_DONE:
```

Comment:
We should probably always check this. The process is saving output fields even before the Data is DONE. We do not want conflicting values in DB.